packages/svelte/src/compiler/compile/Component.js

```diff
@@ -772,12 +772,23 @@ export default class Component {
 			}
 			const writable =
 				node.type === 'VariableDeclaration' && (node.kind === 'var' || node.kind === 'let');
-			const imported = node.type.startsWith('Import');
+
+			let immutable = false;
+			if (node.type === 'VariableDeclaration' && node.kind === 'const') {
+				immutable = true;
+				for (const declaration of node.declarations) {
```

Comment:
I believe in a list of declarations, there's only 1 variable declaration that belongs to the current variable, and that is when the id.name === the name of the variable,
Maybe only then we can opt out the immutable?

Reply:
`id` may not have a name though? Its type is `Pattern` which could be `Identifier | ObjectPattern | ArrayPattern | RestElement | AssignmentPattern | MemberExpression`

Reply:
ok i guess you are right